azerothcore/mod-autobalance
# Divide et impera

## Changes Proposed:
- A single file implementation split into multiple files. Each class has its own file, which allows to split complexity of the module into multiple black boxes that can be better understood and extended in the future.

## Issues Addressed:

- No issues addressed

## SOURCE:


## Tests Performed:

- Compiles with no error and no deprecation warning
- Tested into private server into a dungeon instances; seems to work as old version since no changes has been performed on the logic of the module and balancing.

## How to Test the Changes:


1. Pull the changes
2. Compile the server
3. Enter and complete and instance; i did not touch internal module mechanism except for splitting the classes into multiple files and some decorations.

Take you time to confirm that the module work as intended.
I'm personally performing additional tests to be sure the balancing is not affected. I expect no problem since i just moved code around files and did not touch the code inside the classes and procedures.
